immuni_common/models/mongoengine/batch_file_eu.py

```diff
@@ -0,0 +1,132 @@
+#  Copyright (C) 2020 Presidenza del Consiglio dei Ministri.
+#  Please refer to the AUTHORS file for more information.
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as
+#  published by the Free Software Foundation, either version 3 of the
+#  License, or (at your option) any later version.
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU Affero General Public License for more details.
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import logging
+from datetime import date, datetime, timedelta
+from typing import Dict, List, Optional, Tuple
+
+from bson import ObjectId
+from mongoengine import BinaryField, DateTimeField, Document, EmbeddedDocumentListField, IntField, StringField
+
+from immuni_common.core.exceptions import NoBatchesException
+from immuni_common.models.mongoengine.temporary_exposure_key import TemporaryExposureKey
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BatchFileEu(Document):
+    """
+    Document to wrap a batch of EU-TEKs .
+    """
+
+    index: int = IntField(min_value=0, required=True)
+    keys: List[TemporaryExposureKey] = EmbeddedDocumentListField(
+        TemporaryExposureKey, required=True
+    )
+    period_start: datetime = DateTimeField(required=True)
+    period_end: datetime = DateTimeField(required=True)
+
+    sub_batch_index: int = IntField()
+    sub_batch_count: int = IntField()
+
+    origin: str = StringField(required=True)
```

Comment:
Why doesn't it have a tag like the normal one?

Reply:
because that tag it's an identifier just for the batches that are sent to the gateway and non-viceversa.
We use BatchFile to store the files to be sent to the Italian citizens and to be sent to the European gateway (so the batch_tag it's very important to communicate with them), we use BatchFileEu to store the batches coming from the European gateway so we don't need the batch_tag.

Reply:
It makes sense!
Does the origin have only specific values? If not, why?